src/shared/components/base-question-app.tsx

```diff
@@ -54,6 +56,13 @@ export const BaseQuestionApp = <IAuthoredState extends IAuthoringMetadata & IBas
   useHint();
   useRequiredQuestion();
   useShutterbug({ container: "." + css.runtime });
+  const logging = useBasicLogging({ disabled: disableBasicLogging || !isRuntimeView });
+
+  const setInteractiveStateWithLogging = (updateFunc: UpdateFunc<IInteractiveState>) => {
+    setInteractiveState(updateFunc);
+    // Notify logging helper that answer has been updated by user. Only then it'll be logged on blur.
+    logging.onAnswerUpdate(updateFunc(interactiveState).answerText);
```

Comment:
I think this means the update function is called twice.
I don't have all of this code in my head, but I'd think that might be confusing to an interactive developer when calling setInteractiveState.

But I can't think of a simple way to fix that since we don't know when the updateFunc will be called by setInteractiveState.

Reply:
Yes, it'll be called twice.

But looking at your concerns from other comments, I think I'd vote for removing "answer saved" event and adding "answer_text" to "focus out". In this case, we might even skip this logic and simply always log answer_text on focus out.

Then, LARA could log "answer saved" when the answer is really saved.